### DISCUSSION TITLE
👀  Important Updates: Copilot Chat API endpoints and Copilot Content Exclusions

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#101438] [Copilot News and Announcements]
👋 We have three key GitHub Copilot updates from this week to share with you. 

## 1️⃣ Copilot Content Exclusions Feature Update 🔐 

Following our [previous](https://github.blog/changelog/2023-11-08-copilot-content-exclusion-is-now-available-in-public-beta) communication dated November 8, 2023, regarding the temporary rollback of the Copilot content exclusions feature, we are pleased to announce the re-deployment of this feature with significant enhancements. The rollout will be progressive during the next 10 days as we monitor the behaviour.

With Content Exclusion, **GitHub Copilot Business customers will be able to prevent specified files or repositories from being used to inform code completion suggestions made by GitHub Copilot**. GitHub Copilot will not be available in excluded files. Organization administrators or repository owners can choose which files or repositories are excluded. [Learn more](https://docs.github.com/en/copilot/managing-copilot-business/configuring-content-exclusions-for-github-copilot).

### Overview of the Issue 👀 

Our team observed a critical issue where clients were incorrectly blocked from using Copilot due to the initial implementation of content exclusions. This was primarily caused by errors in fetching content exclusion policies from the client, leading to a temporary suspension of the feature.

### Actions Undertaken 🏁 

In response to this, our engineering team undertook a comprehensive review and rectification process. The issues identified in the client's code were addressed, and additional verifications were implemented on both server and client sides to prevent recurrence.

### New Enhancements in the Re-deployed Feature 🆕 

*   **Performance Update**: We have optimized the performance of the content exclusions feature, ensuring minimal impact on the user experience.
*   **Extended Coverage**: The feature now supports all our official Integrated Development Environments: Visual Studio, JetBrains IDEs, Visual Studio Code, and Vim/Neovim.

### Current Status 🟢 

*   Users with pre-existing content exclusion configurations will experience no change.
*   New and returning users can now utilize the enhanced feature across all supported IDEs.

**Up Next** ❓  We are closely monitoring the performance and user feedback post-deployment. The support for Copilot Chat is also in progress and will be part of the General Availability.

## 2️⃣ Upcoming deprecation of Copilot Chat API endpoints   🌐 

***
**Note**: If you are using the latest version of the Copilot extension for Visual Studio or VS Code or you are using the Copilot plugin for JetBrains IDEs, you will not be impacted by this change.
***

As we [announced in September](https://github.blog/changelog/2023-09-25-breaking-change-to-github-copilot-service-endpoints/), we updated the API service endpoints used by Copilot Chat. **On February 1, 2024, we will deprecate the Copilot Chat API endpoints** currently being routed through `https://copilot-proxy.githubusercontent.com`. Instead, these requests will go through `https://api.githubcopilot.com`. Versions 0.8.0 and later of the Copilot Chat extension for VS Code and versions 0.1.1817.27579 and later of the Copilot Chat extension for Visual Studio already route chat traffic through `https://api.githubcopilot.com`. All versions of the Copilot plugin for JetBrains IDEs already route chat traffic through `https://api.githubcopilot.com`.

_**To ensure Copilot Chat continues working from **February 1, 2024**, please update to the most recent version of the Copilot extension and ensure your firewall and network settings allow communication to `https://api.githubcopilot.com`.**_

Also, note that as of January 19, 2024, we have closed the Copilot Enterprise waitlist in anticipation of the general availability release of Copilot Enterprise.

## 3️⃣  Update on the Copilot Enterprise Waitlist ⏫

On January 19, 2024, we will close the Copilot Enterprise waitlist in anticipation of the general availability release of Copilot Enterprise.

Got questions or feedback to share? Join us in the [community](https://github.com/orgs/community/discussions/categories/copilot) or comments below.

## Other important updates from January ⏫ 

- [Copilot Enterprise beta – January 10th update](https://github.blog/changelog/2024-01-10-whats-new-in-copilot-enterprise-beta-january-10th-update/)
- [GitHub Copilot Chat now generally available for organizations and individuals](https://github.blog/changelog/2024-01-11-github-copilot-chat-now-generally-available-for-organizations-and-individuals/)

**Comment by cmp-nct** [comment]
whatever you did in the past 3-4 weeks, copilot is worse than it was at launch.
I can't recall the last chat response I received that was actually containing anything I could partially use.
The open source code models start to approach GPT-4 level, sometimes surpassing it. Mostly because GPT4 degraded to Turbo-3.5 dumbness but also because they improved quite nicely. 

I've switched to continue.dev (after maybe a thousand hours copilot use), that's how I could at least continue developing after I got used to what copilot was in the past.

At this point it's quite obvious that OpenAI LLM supremacy is going to be lost in the next months, partially from self inflicted wounds and partially because their pace of innovation is crippled by internal struggles. Maybe also by losing some important devs, hard to tell but the conclusion seems set in stone.

**Comment by verbeemen** [comment]
> Up Next ❓ We are closely monitoring the performance and user feedback post-deployment. The support for Copilot Chat is also in progress and will be part of the General Availability.

Does this mean that there will be an exclusion list as well for the copilot's chat function?

**Reply by Aiphonec** [reply]
No

**Comment by oldhand7** [comment]
Hello, everyone. I am a AI & ML developer and I am really interested in your state of art work. Thanks for your great work. I would like to share my opinion about your work. I noticed that your version doesn't support multiple thread so that multiple users can't get a service at the same time. They have to wait until previous one had completed. I wonder if it will be possible to tackle it. Looking forward to your answer.

**Comment by smooth-web** [comment]
[smooth-web-main.zip](https://github.com/smooth-web/smooth-web/files/14201959/smooth-web-main.[smooth-web-main.zip](https://github.com/smooth-web/smooth-web/files/14202398/smooth-web-main.zip))
- 👋 Hi, I’m @smooth-web
- 👀 I’m interested in ...
- 🌱 I’m currently learning ...
- 💞️ I’m looking to collaborate on ...
- 📫 How to reach me ...
- 😄 Pronouns: ...
- ⚡ Fun fact: ...

**Comment by Hilbert1984** [comment]
I'm getting the following error when I try to use GitHub Copilot Chat on VS Studio Code: error [error] [auth] Extension activation failed: "read ECONNRESET". I think its indeed related to the Copilot Chat API endpoints issue because I get an error 404 page not found for https://[api.githubcopilot.com](https://api.githubcopilot.com/). I have the latest version of Visual Studio Code and also the Github Copilot extension but its still not solving the issue.

**Comment by trilebh** [comment]
is there any solution for fixing this issue? the API copilot is down and i am getting error. Please help!

[INFO] [default] Error sending telemetry FetchError: read ECONNRESET
    at fetch (C:\snapshot\copilot-client\node_modules\@adobe\helix-fetch\src\fetch\index.js:99:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at cachingFetch (C:\snapshot\copilot-client\node_modules\@adobe\helix-fetch\src\fetch\index.js:288:16)
    at cX.fetch (C:\snapshot\copilot-client\lib\src\network\helix.ts:78:22) {
  type: 'system',
  _name: 'FetchError',
  code: 'ECONNRESET',
  errno: -4077,
  erroredSysCall: 'read'

**Comment by Saa1999662005** [comment]
https://ceac.state.gov
